onetl/connection/file_connection/samba.py

```diff
@@ -0,0 +1,306 @@
+                raise ConnectionError("Failed to connect to the Samba server.")
+        except (RuntimeError, ValueError):
+            # left validation errors intact
+            log.exception("|%s| Connection is unavailable", self.__class__.__name__)
+            raise
+        except Exception as exc:
+            log.exception("|%s| Connection is unavailable", self.__class__.__name__)
+            raise RuntimeError("Connection is unavailable") from exc
+
+        return self
+
+    @slot
+    def path_exists(self, path: os.PathLike | str) -> bool:
+        try:
+            self.client.getAttributes(self.share, str(path))
+            return True
+        except OperationFailure:
+            return False
+
+    def _scan_entries(self, path: RemotePath) -> list:
+        if self._is_dir(path):
+            return [
+                entry
+                for entry in self.client.listPath(
+                    self.share,
+                    str(path),
+                )
+                if entry.filename not in {".", ".."}  # Filter out '.' and '..'
+            ]  # pysmb replaces '/', not works with <RemotePath> type
+        return [self.client.getAttributes(self.share, (os.fspath(path)))]
+
+    def _extract_name_from_entry(self, entry) -> str:
+        return entry.filename
+
+    def _is_dir_entry(self, top: RemotePath, entry) -> bool:
+        return entry.isDirectory
+
+    def _is_file_entry(self, top: RemotePath, entry) -> bool:
+        return not entry.isDirectory
+
+    def _extract_stat_from_entry(self, top: RemotePath, entry) -> RemotePathStat:
+        if entry.isDirectory:
+            return RemotePathStat(st_mode=stat.S_IFDIR)
+
+        return RemotePathStat(
+            st_size=entry.file_size,
+            st_mtime=entry.last_write_time,
+            st_uid=entry.filename,
+        )
+
+    def _get_client(self) -> SMBConnection:
+        is_direct_tcp = self.protocol == "SMB"
+        use_ntlm_v2 = self.auth_type == "NTLMv2"
+        conn = SMBConnection(
+            username=self.user,
+            password=self.password.get_secret_value() if self.password else None,
+            my_name="optional_client_name",
+            remote_name=self.host,
+            domain=self.domain,
+            use_ntlm_v2=use_ntlm_v2,
+            sign_options=2,
+            is_direct_tcp=is_direct_tcp,
+        )
+        conn.connect(self.host, port=self.port)
+        return conn
+
+    def _is_client_closed(self, client: SMBConnection) -> bool:
+        try:
+            client.listShares()
+        except NotConnectedError:
+            return True
+        return False
+
+    def _close_client(self, client: SMBConnection) -> None:
+        self.client.close()
+
+    def _download_file(self, remote_file_path: RemotePath, local_file_path: LocalPath) -> None:
+        with open(local_file_path, "wb") as local_file:
+            self.client.retrieveFile(
+                self.share,
+                str(remote_file_path),
```

Comment:
```suggestion
                os.fspath(remote_file_path),
```
Во всех остальных местах тоже нужно заменить `str(path)` на `os.fspath(path)`